### ISSUE TITLE
Workshop Schedule

### ISSUE THREAD

**Issue by miykael** [miykael/workshop_pybrain#4]
Hi @peerherholz,

I'm trying to figure out the best schedule for the workshop, so that it's not overloaded nor too boring. But I'm still missing a few puzzle pieces. See here:

# General thoughts

- Learners will be participating in our course everyday from 09:30 to 12:00 and from 13:00 to 15:30. This means we have four half days at 2.5h to cover. Luckily we have much more content to offer.
- Our workshop is called "MRI analysis in Python using Nipype, Nilearn and more". Given this, I'm curious if we should keep the "Docker & Conda" or "Virtualization" part to a minimum.
- Given that there's a Dipy workshop two weeks later, we can drop our Dipy section

# Program Suggestion

## Day 1 - Monday, 9th November

**Morning** - @miykael
- `30min` - **Workshop intro** (who are we, what will we cover, where can content be found, which workshop tools will we use? sli.do, ...
- `60min` - **Intro/Recap** to Python, Jupyter Notebooks and Scientific Python (mostly show and tell)
- `60min` - **Nibabel & Nilearn** (~30min exercises)

**Afternoon** - @peerherholz
- `30min` - **Conda & Docker** (just show and tell)
- `45min` - **Resting State / Connectivity** (with exercises)
- `45min` - **Your ML intro** with classification of connectivity matrices
- `30min` - **Slides: Innovation in Field - Part I**

## Day 2 - Tuesday, 10th November

**Morning** - @miykael
- `30min` - **Nipype intro slides & Nipype Showcase**
- `60min` - **Nipype basics**: Interfaces, Nodes, Workflows, Data I/O, Iterations, Errors, Debugging (with 30min exercises)
- `60min` - **Nipype Workflow Example** - Hands-On 1 Preprocessing (with 30min exercises)

**Afternoon** - @peerherholz
- `20min` - **PyBIDS**
- `40min` - **Nistats** (with exercises)
- `15min` - **ML preparation**
- `75min` - **Searchlight & Keras** (Nilearn, pymvpa, Keras)
- `30min` - **Slides: Innovation in Field - Part II**

# Comments

Like this the first three half-days would work nicely, but the last afternoon is too packed (i.e. 3h instead of 2.5h). My potential solutions are:
- Moving PyBIDS to another section or even dropping it? It's halfway covered in the Nipype I/O part.
- Searchlight section is still rather charged. We could drop Keras. I've never used it myself. Classification of connectivity matrix and searchlight seems much more important.
- **Side note**: The **Innovation in Field** slides are reduced to only 60min in this version. Do you think this is too tight?

What are your thoughts? It's all still very dense. But I personally believe that it gives the participants a lot of content to go back to later. Nonetheless, it might be worth it to drop some stuff in the slides or notebooks.

**Comment by PeerHerholz** [comment]
Ahoi hoi @miykael,

thanks a lot for this! 
The basics: I'll try to join around 12:30 PM workshop time zone, thus can't really help re everything before in terms of active involvement during the workshop. I mean I could (with a lot of coffee), if needed.

- `conda` and `containers` should stay at a minimum, I agree

- `resting state/connectivity` as we did so far, right? I like the continuation of this topic into ML

- `keras` or `pymvpa` could be dropped...Given that `pymvpa` is less and less used and folks are very into `DL` these days, I would actually keep `keras` (or maybe adapt the example for `tensorflow`). What could/should be worth including is `ROI decoding` as I don't think we have that yet, eh?

- with that we could keep `pybids` 

- I keep thinking about other tools we didn't include so far...a short intro to meta-analyses using [nimare](https://github.com/neurostuff/NiMARE) and gradients using [brainspace](https://github.com/MICA-MNI/BrainSpace) would be great additions and state-of-the-art

- cutting down `innovations in the field` should be ok, as a lot more people have heard of/use `BIDS` and friends since we started this 2 1/2 years ago, hehe

LMKWYT

**Comment by miykael** [comment]
No worries about not joining in the morning. First morning is anyhow intro and second is Nipype. So all fine for me. My point was more, don't hesitate to tell me if you want to give me some things in the afternoon. Happy to switch it a bit if you'd like (or also to lay back ;-))

- Yes, `conda`/`containers` should stay. Otherwise explaining BIDSapp is difficult
- `resting state/connectivity` stays as is, and exactly, would allow a smooth transition into "ML intro"
- I will drop `keras` and `pymvpa` but on the side explore a short `keras/tensorflow` setup. Perhaps with a pretrained model.
- Great, `pybids` will stay.
- `innovations in the field` down to 60min sounds good. Do you want to trim it down or we leave the slides and just feel happy to skip?
- Additional tools like `nimare` and `brainspace` sound cool. Unfortunately I don't know them, nor have the capacity to setup a notebook. What about adding them to the "innovations in the fields" slides? Is perhaps a quicker solution?

**Comment by PeerHerholz** [comment]
Coolio, I think we're getting closer!

- if `keras` and `pymvpa` will be drop, we can also drop the `preparation` and would ultimately get at least 30 min free minutes, no? That's assuming we include an ROI decoding example

- do you think 60 min for the `python recap` are necessary given that a comprehensive introduction will be taught a bit before? Maybe this could be shortened to 30 min?

- I think the slides could remain as they're right now and we just go through some things quicker. TBH, I might check a conversion to google slides as this might be easier for a lot of folks

- I could check `nimare` and `brainspace` as both have great tutorials already, thus we could just put some things together, but I think I could do that earliest next week

**Comment by miykael** [comment]
I've updated the program (see [here on nbviewer](https://nbviewer.jupyter.org/github/miykael/workshop_pybrain/blob/master/program.ipynb). I made everything more color full (which is hopefully useful, but let me know if glows like a Christmas tree).

I haven't cleaned out the slides or notebooks yet, but I gave each half day 3-4 sections, what do you think?

---

**Schedule - Day 1</h3>**  
- `09:30-10:00` Welcome, and workshop overview</h4>
- `10:00-10:30` Quick recap on Jupyter, Python and more</h4>
- `10:30-12:00` Explore MRI data with Nibabel and Nilearn</h4>
- `12:00-13:00` Lunch</h4>
- `13:00-13:30` How to set up your system, using Conda and Docker</h4>
- `13:30-15:00` Functional connectivity and machine learning</h4>
- `15:00-15:30` Innovations in neuroimaging tools (Part I)</h4>
- `15:30-...` Open ended for questions</h4>

**Schedule - Day 2</h3>**  
- `09:30-10:00` Introduction to Nipype</h4>
- `10:00-11:00` Exploration of Nipype's moving parts</h4>
- `11:00-12:00` Creation of a Nipype Pipeline</h4>
- `12:00-13:00` Lunch</h4>
- `13:00-14:00` PyBIDS and Nistats</h4>
- `14:00-15:00` Multivariate pattern analysis using Searchlight and Deep Learning</h4>
- `15:00-15:30` Innovations in neuroimaging tools (Part 2)</h4>
- `15:30-...` Open ended for questions</h4>

**Comment by miykael** [comment]
To your previous comments:

1. I dropped `Keras` and `PyMVPA`. I will explore another deep learning model (using pretrained 3D-RESNET that I saw already in Keras challenges), but I might drop them as I'm not expert in them.
2. What do you mean with ROI decoding example? Searchlight or something else?
3. You're right, `python recap` can be shortened to 30min. Many people will already know it or have taken the previous workshop 👍 
4. To familiarize myself again with the slides I wanted to give them a quick look. I might also adapt a few (like the Python recap), but yes. They can mostly stay the same. I'm ok with conversion to google slides. Is there a quick way to do this?
5. With `nimare` and `brainspace`, be my guest. I think we still have time to develop things.

What I would like to fix **asap** is a timetable, so that it can also be communicated to learners. Similar to the first workshop [here](https://pybrain-workshop.github.io/).

**Comment by PeerHerholz** [comment]
Cool, thx!

1. sounds about right. 

2. e.g. can we decode faces vs. houses in the PPA? here, we could include permutation tests as well.

3. cool.

4. I think now, one would have to go through the slides and adapt them. However, I'll check that.

5. Ok, I'll look into it.

I think the one you outlined above could be shared, no?

**Comment by miykael** [comment]
Perfect, yes. I will share this with the organizers. The Docker installation instructions were also updated and so everything should be good to go. Notebooks and slides can be changed for quite some time, everything is updated with a quick `docker pull`.

With respect to "2. e.g. can we decode faces vs. houses in the PPA? here, we could include permutation tests as well." Do you have any specific pointers or code snippets? Happy to flesh it out if need be, but I'm not sure what you're pointing at :-)

**Comment by PeerHerholz** [comment]
Re `ROI decoding` I was referring to this [example](https://nilearn.github.io/auto_examples/02_decoding/plot_haxby_full_analysis.html#sphx-glr-auto-examples-02-decoding-plot-haxby-full-analysis-py). I'll add further stuff in the other issue.

**Comment by miykael** [comment]
Issue can be closed, ROI decoding will be integrated with the next commit.
